providers/default.rb

```diff
@@ -202,7 +202,11 @@ def run_post_install_action
       end
     end
     Chef::Log.warn 'Replacing chef-client process with upgraded version and re-running.'
-    Kernel.exec(new_resource.exec_command, *new_resource.exec_args)
+    env = {}
```

Comment:
What's this bit for?

Reply:
As the first arg to `Kernel.exec` we can optionally specify the environment - http://ruby-doc.org/core-2.4.5/Kernel.html#method-i-spawn

This is how I am setting the CHEF_LICENSE since this cookbook does not manage the client.rb